rust/lancedb/src/connection.rs

```diff
@@ -133,7 +133,7 @@ impl IntoArrow for NoData {
 
 /// A builder for configuring a [`Connection::create_table`] operation
 pub struct CreateTableBuilder<const HAS_DATA: bool, T: IntoArrow> {
-    parent: Arc<dyn ConnectionInternal>,
+    pub(crate) parent: Arc<dyn ConnectionInternal>,
```

Comment:
needed? seems like a leaky abstraction?

Reply:
I feel like we do need to expose this, so we can use it to construct OpenTableBuilder which gets passed to the callback of `CreateTableMode::ExistOK(<callback>)`.